packages/redux-devtools-extension/src/index.ts

```diff
@@ -195,8 +195,10 @@ interface ReduxDevtoolsExtension {
 }
 
 export interface ReduxDevtoolsExtensionCompose {
-  (config: Config): (...funcs: StoreEnhancer[]) => StoreEnhancer;
-  (...funcs: StoreEnhancer[]): StoreEnhancer;
+  (config: Config): <StoreExt>(
+    ...funcs: Array<StoreEnhancer<StoreExt>>
+  ) => StoreEnhancer<StoreExt>;
+  <StoreExt>(...funcs: Array<StoreEnhancer<StoreExt>>): StoreEnhancer<StoreExt>;
```

Comment:
Wouldn't this make it so that all of the enhancers would need to have the `StoreExt` generic? That doesn't seem ideal, although it still might be better than the status quo.